common/utils/src/main/scala/org/apache/spark/util/MavenUtils.scala

```diff
@@ -335,10 +335,12 @@ private[spark] object MavenUtils extends Logging {
 
   /* Set ivy settings for location of cache, if option is supplied */
   private def processIvyPathArg(ivySettings: IvySettings, ivyPath: Option[String]): Unit = {
-    ivyPath.filterNot(_.trim.isEmpty).foreach { alternateIvyDir =>
-      ivySettings.setDefaultIvyUserDir(new File(alternateIvyDir))
-      ivySettings.setDefaultCache(new File(alternateIvyDir, "cache"))
+    val alternateIvyDir = ivyPath.filterNot(_.trim.isEmpty).getOrElse {
+      // To protect old Ivy-based systems like old Spark from Apache Ivy 2.5.2's incompatibility.
+      System.getProperty("user.home") + File.separator + ".ivy2.5.2"
```

Comment:
Is ".ivy2.5.2" too specific? like would the new ivy dir be used, probably, by subsequent ivy versions? maybe ".ivy2.5" or something. It doesn't really matter though.

Reply:
Ya, we can do that. I also thought `.ivy2.5` first, but it was misleading because Ivy 2.5.0 and 2.5.1 are incompatible with 2.5.2. I want to say `2.5.2+` here, but I dropped `+`.

Reply:
Sounds good to me